test/e2e/specs/splitting-merging.test.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Internal dependencies
+ */
+import '../support/bootstrap';
+import { newPost, newDesktopBrowserPage } from '../support/utils';
+
+describe( 'splitting and merging blocks', () => {
+	beforeAll( async () => {
+		await newDesktopBrowserPage();
+		await newPost();
+	} );
+
+	it( 'Should split and merge paragraph blocks using Enter and Backspace', async () => {
+		//Use regular inserter to add paragraph block and text
+		await page.click( '.edit-post-header [aria-label="Add block"]' );
+		await page.keyboard.type( 'paragraph' );
+		await page.keyboard.press( 'Tab' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'FirstSecond' );
+
+		//Move caret between 'First' and 'Second' and press Enter to split paragraph blocks
+		for ( let i = 0; i < 6; i++ ) {
+			await page.keyboard.press( 'ArrowLeft' );
+		}
+		await page.keyboard.press( 'Enter' );
+
+		//Switch to Code Editor to check HTML output
+		await page.click( '.edit-post-more-menu [aria-label="More"]' );
+		let codeEditorButton = ( await page.$x( '//button[contains(text(), \'Code Editor\')]' ) )[ 0 ];
+		await codeEditorButton.click( 'button' );
+
+		//Assert that there are now two paragraph blocks with correct content
+		let textEditorContent = await page.$eval( '.editor-post-text-editor', ( element ) => element.value );
+		expect( textEditorContent ).toMatchSnapshot();
```

Comment:
Not related to this PR, but I feel like we have a bunch of assertions like this:

 - Move to the text Editor
 - check the content with a snapshot
 - Move back to the visual editor

maybe we should create an utility for that?